components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect, useRef, useMemo } from "react";
+import styles from "./AssignmentTimer.module.css";
+
+interface Props extends React.PropsWithChildren {
+  durationInMinutes: number;
+  onTimerEnd?: () => void;
+}
+
+const timeFormatter = (totalSeconds: number) => {
+  const totalSecondsToHours = Math.floor(totalSeconds / 3600);
+  const totalSecondsToMinutes = Math.floor((totalSeconds % 3600) / 60);
+  const totalSecondsToSeconds = Math.floor((totalSeconds % 3600) % 60);
+
+  const hours = totalSecondsToHours.toString().padStart(2, "0");
+  const minutes = totalSecondsToMinutes.toString().padStart(2, "0");
+  const seconds = totalSecondsToSeconds.toString().padStart(2, "0");
+
+  return `${hours}:${minutes}:${seconds}`;
+};
+
+const AssignmentTimer: React.FC<Props> = ({
+  durationInMinutes,
+  onTimerEnd,
+}) => {
+  const timerRef = useRef<NodeJS.Timer>();
+  const [timeRemaining, setTimeRemaining] = useState(() => durationInMinutes * 60);
```

Comment:
Hello, rashad. I'm a bit curious, if I may ask, the time remaining unit used here is seconds, cmiiw? Let me know 😁

Reply:
Hello Alex, yes it is in seconds. From consulting with the backend team, they will send the duration in minutes:D